### PR TITLE
beer_styles: store data as JSON

### LINKED ISSUE #43: Local database support
As it is very painful to keep the beer, hops and yeast data in Rust structs, find out if there is optional solution to store this data inside the executable/lib. For example file based database that could travel inside the package. See #23 #39 #41 #42 for use cases of this.

### PR DESCRIPTION
**Pull request by tommilligan**
Closes #43

Implements the `beer_styles` data as a JSON data file, rather than native rust structs.

This data is inlined at compile time using the `include_str!` macro, and deserialised the first time `BEER_STYLES` is dereferenced (using `once_cell`).

The additional refactoring of `commands/beer_style.rs` is due to the fact that `BeerStyle` is no longer `Copy` as it contains a `String`. The way results are calculated now results in less data copying overall, so I think it's a win still.

> is it possible to do somehow so that the users of the library have the same data available?

Yep, external users can still use the data as before by `use rustybeer::utils::beer_styles::BEER_STYLES;`. The type has changed to `Vec<BeerStyle>` though. If you'd already published `0.1.0` that would make this a breaking change.

### CODE REVIEW

**Comment by drodil** [comment]
Thanks for the contribution! We haven't pushed this package yet as it's missing quite a lot of things before being useful for anybody. I will check the changes when I get back on computer

File: src/commands/beer_style.rs [hunk 1]
```diff
@@ -48,39 +48,25 @@ pub fn add_subcommand<'a, 'b>() -> App<'a, 'b> {
         )
 }
 
+fn matches_to_criteria<'a>(matches: &ArgMatches<'a>) -> Criteria {
+    Criteria {
+        og: matches.value_of("og").map(|value| value.parse().unwrap()),
```
Comment (drodil) [inline src/commands/beer_style.rs:53]: Ain't these exact matches? Idea was to find beer styles where input of for example OG is between the beer styles min and max OG.

**Review by drodil — COMMENTED** [review]
Looks good but wondering about the matches for beer styles in the command

File: src/commands/beer_style.rs (hunk 1 above)
Comment (tommilligan) [reply src/commands/beer_style.rs:53]: This function is just responsible for parsing the command line options into numbers and returning a `Criteria` struct - the actual range matching logic still exists in the `utils/beer_styles.rs` file, and behaves the same as before with ranges:

https://github.com/drodil/rustybeer/blob/d336213ad27e52162b936de4a872dcf1f5a8fc2d/src/utils/beer_styles.rs#L42

**Comment by tommilligan** [comment]
Clippy appears to be failing with some existing master issues

**Review by drodil — APPROVED** [review]

**Comment by drodil** [comment]
@all-contributors please add @tommilligan   for Code

**Pull request merged**